transitions/index.js

```diff
@@ -103,6 +117,83 @@ const processChange = (change, callback) => {
     });
 };
 
+const deleteInfoDoc = (change, callback) => {
+  db.medic.get(`${change.id}-info`, (err, doc) => {
+    if (err) {
+      if (err.statusCode === 404) {
+        // don't worry about deleting a non-existant doc
+        return callback();
+      }
+      return callback(err);
+    }
+    doc._deleted = true;
+    db.medic.insert(doc, callback);
+  });
+};
+
+const getAdminNames = callback => {
+  const options = { key: [ 'user-settings' ], include_docs: true };
+  db.medic.view('medic-client', 'doc_by_type', options, (err, results) => {
+    if (err) {
+      return callback(err);
+    }
+    const admins = results.rows
+      .filter(row => {
+        return row.doc.roles &&
+               (row.doc.roles.includes('_admin') ||
+               row.doc.roles.includes('national_admin'));
+      })
+      .map(row => {
+        const parts = row.id.split(':');
+        return parts.length > 1 && parts[1];
+      })
+      .filter(name => !!name);
+    callback(null, admins);
+  });
+};
+
+const deleteReadDocs = (change, callback) => {
+
+  // we don't know if the deleted doc was a report or a message so
+  // attempt to delete both
+  const possibleReadDocIds = [ 'report', 'message' ]
+    .map(type => `read:${type}:${change.id}`);
+
+  getAdminNames((err, admins) => {
```

Comment:
Do we have access to the entire document at this stage? Ideally we'd only do this if the document type was `data_record`.

Reply:
Unfortunately, no. Just the id. And we can't fetch it either because it's just been deleted.